### ISSUE TITLE
PolicyKey Behavior in PolicyWrap 

### ISSUE THREAD

**Issue by andregrohmann** [App-vNext/Polly#463]
Hi,

i just started with Polly and need some clarification about the behavior of PolicyKeys in a PolicyWrap.
According to [Keys-and-Context-Data](https://github.com/App-vNext/Polly/wiki/Keys-and-Context-Data) the PolicyKey should retain its own value inside a PolicyWrap.
Consider the following example:
```
[Fact]
public void PolicyKeysInPolicyWrap()
{
    var fallback = Policy<string>
        .Handle<Exception>()
        .Fallback<string>("FallbackValue", onFallback: (result, context) =>
        {
            context.PolicyWrapKey.Should().Be("PolicyWrap");
            context.PolicyKey.Should().Be("FallbackPolicy"); //but is "RetryPolicy"
        })
        .WithPolicyKey("FallbackPolicy");

    var retry = Policy<string>
        .Handle<Exception>()
        .Retry(1, onRetry: (result, retryCount, context) =>
        {
            context.PolicyWrapKey.Should().Be("PolicyWrap");
            context.PolicyKey.Should().Be("RetryPolicy");
        })
        .WithPolicyKey("RetryPolicy");

    var policyWrap = Policy.Wrap(fallback, retry)
        .WithPolicyKey("PolicyWrap");

    policyWrap.Execute(ThrowException);

    string ThrowException()
    {
        throw new Exception();
    }
}
```
In `onFallback` i expected the PolicyKey to be the key of fallback policy and not the key of retry policy.
Is this a Bug or have i simply gotten it wrong.

Thanks in advance.

**Comment by reisenberger** [comment]
@andregrohmann This looks like a bug: the `PolicyKey` should revert to the outer policy, as you say.  I have a plan for a fix for which the implementation would tie in nicely with work on Polly metrics.  Thanks for raising this.

**Comment by andregrohmann** [comment]
@reisenberger  Thank you for the clarification.

**Comment by reisenberger** [comment]
Will be fixed with the release of v6.1.0. Fixed in #485

**Comment by reisenberger** [comment]
Thank you @andregrohmann for raising this! Fixed in v6.1.0 which will release to nuget shortly
